File: Day21/Frontend/src/services/product.service.js
import axios from "axios";

const API_URL = "http://localhost:8082/api/products/";

const addProduct = (product) => {
  return axios.post(API_URL + "add", product);
};

const getAllProducts = () => {
  return axios.get(API_URL + "all");
};

const getProductById = (id) => {
  return axios.get(API_URL + id);
};

const updateProduct = (id, product) => {
  return axios.put(API_URL + id, product);
};

const deleteProduct = (id) => {
  return axios.delete(API_URL + id);
};

const ProductService = {
  addProduct,
  getAllProducts,
  getProductById,
  updateProduct,
  deleteProduct,
};

export default ProductService;

File: Day22/Backend/stockx/src/main/java/com/guruprasath/stockx/service/ProductService.java
package com.guruprasath.stockx.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.guruprasath.stockx.model.Product;
import com.guruprasath.stockx.repository.*;

import java.util.List;
import java.util.Optional;

@Service
public class ProductService {

    private final ProductRepository productRepository;

    @Autowired
    public ProductService(ProductRepository productRepository) {
        this.productRepository = productRepository;
    }

    public List<Product> getAllProducts() {
        return productRepository.findAll();
    }

    public Product getProductById(Long id) {
        Optional<Product> productOptional = productRepository.findById(id);
        return productOptional.orElse(null);
    }

    public Product createProduct(Product product) {
        return productRepository.save(product);
    }

    public Product updateProduct(Long id, Product updatedProduct) {
        Optional<Product> productOptional = productRepository.findById(id);
        if (productOptional.isPresent()) {
            updatedProduct.setId(id);
            return productRepository.save(updatedProduct);
        }
        return null; // Product not found
    }

    public void deleteProduct(Long id) {
        productRepository.deleteById(id);
    }
}
